test/unit/agent/agent.test.js

```diff
@@ -16,643 +19,562 @@ const Transaction = require('../../../lib/transaction')
-  t.autoend()
-
-  const origInterval = global.setInterval
+test('initial harvest', async (t) => {
+  t.beforeEach(async (ctx) => {
+    ctx.nr = {}
 
-  let agent = null
-  let redirect = null
-  let connect = null
-  let settings = null
-
-  t.beforeEach(() => {
-    nock.disableNetConnect()
-
-    global.setInterval = (callback) => {
-      return Object.assign({ unref: () => {} }, setImmediate(callback))
-    }
+    const collector = new Collector()
+    ctx.nr.collector = collector
+    await collector.listen()
 
-    // Load agent with default 'stopped' state
-    agent = helper.loadMockedAgent(null, false)
-
-    // Avoid detection work / network call attempts
-    agent.config.utilization = {
+    ctx.nr.agent = helper.loadMockedAgent(collector.agentConfig, false)
+    ctx.nr.agent.config.utilization = {
       detect_aws: false,
       detect_pcf: false,
       detect_azure: false,
       detect_gcp: false,
       detect_docker: false
     }
-
-    agent.config.no_immediate_harvest = true
-
-    redirect = nock(URL)
-      .post(helper.generateCollectorPath('preconnect'))
-      .reply(200, {
-        return_value: {
-          redirect_host: 'collector.newrelic.com',
-          security_policies: {}
-        }
-      })
-
-    connect = nock(URL)
-      .post(helper.generateCollectorPath('connect'))
-      .reply(200, { return_value: { agent_run_id: RUN_ID } })
-
-    settings = nock(URL)
-      .post(helper.generateCollectorPath('agent_settings', RUN_ID))
-      .reply(200, { return_value: [] })
+    ctx.nr.agent.config.no_immediate_harvest = true
   })
 
-  t.afterEach(() => {
-    global.setInterval = origInterval
-
-    helper.unloadAgent(agent)
-    agent = null
-
-    if (!nock.isDone()) {
-      /* eslint-disable-next-line no-console */
-      console.error('Cleaning pending mocks: %j', nock.pendingMocks())
-      nock.cleanAll()
-    }
-
-    nock.enableNetConnect()
+  t.afterEach((ctx) => {
+    helper.unloadAgent(ctx.nr.agent)
+    ctx.nr.collector.close()
   })
 
-  t.test('should not blow up when harvest cycle runs', (t) => {
+  await t.test('should not blow up when harvest cycle runs', async (t) => {
+    const { promise, resolve } = promiseResolvers()
+    const { agent, collector } = t.nr
     agent.start(() => {
       setTimeout(() => {
-        t.ok(redirect.isDone())
-        t.ok(connect.isDone())
-        t.ok(settings.isDone())
-
-        t.end()
+        assert.equal(collector.isDone('preconnect'), true)
+        assert.equal(collector.isDone('connect'), true)
+        assert.equal(collector.isDone('agent_settings'), true)
+        resolve()
       }, 15)
     })
+    await promise
   })
 
-  t.test('should start aggregators after initial harvest', (t) => {
+  await t.test('should start aggregators after initial harvest', async (t) => {
+    const { promise, resolve } = promiseResolvers()
+    const { agent, collector } = t.nr
+
     sinon.stub(agent.harvester, 'start')
+    t.after(() => sinon.restore())
```

Comment:
This `sinon.restore()` is the sole reason for updating to `sinon@5` (and evidently `sinon@18` is the current version). Basically, in `sinon@5` the default import is itself a "sandbox" and thus obviates the need to `const sandbox = sinon.sandbox()` and do a dance around that for this simple test.